### PR TITLE
[spotify-api] Typings for shows, episodes and market endpoints

### PR DESCRIPTION
**Pull request by eegli**
- Add typings for [save](https://developer.spotify.com/documentation/web-api/reference/#/operations/save-episodes-user ), remove and check shows for user as well as for [get available markets](https://developer.spotify.com/documentation/web-api/reference/#/operations/get-available-markets)
- Add response for [get the user's queue](https://developer.spotify.com/documentation/web-api/reference/#/operations/get-queue)
- Add response for [get playlist cover image](https://developer.spotify.com/documentation/web-api/reference/#/operations/get-playlist-cover)
- Add missing property `html_description` to `ShowObjectSimplified` 
- Cleanup tests - by shortening the `available_markets` property from most objects, we can save a few thousand lines
--- 
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by piksel** [comment]
Did you really mean to remove 7k from the tests?

File: types/spotify-api/index.d.ts [hunk 1]
```diff
@@ -263,10 +263,13 @@ declare namespace SpotifyApi {
     }
 
     /**
-     * Get audio analysis for a track
+     * Get Audio Analysis for a Track
      *
      * GET /v1/audio-analysis/{id}
-     * https://developer.spotify.com/documentation/web-api/reference/#/operations/get-audio-analysis
+     * https://developer.spotify.com/web-api/get-audio-analysis/
```
Comment (piksel) [inline types/spotify-api/index.d.ts:269]: It's a bit annoying that some documentation is available using the "short form" (which is much nicer), as this leads to the different documentation links looking non-uniform. Having ones without any anchor hash is much better for the developer experience :/

**Review by piksel — APPROVED** [review]
Yeah, looks good. Reducing the contents of the responses in the test body was absolutely a good catch. It may not be as accurate, but it does not alter what we're testing in any significant way.

File: types/spotify-api/index.d.ts (hunk 1 above)
Comment (eegli) [reply types/spotify-api/index.d.ts:269]: It appears that they only support the redirect from the short form on "old" endpoints. E.g.:
- https://developer.spotify.com/web-api/get-audio-analysis works
- https://developer.spotify.com/web-api/save-shows-user does not

We can't really do anything about it, unfortunately.

There's also https://developer.spotify.com/web-api/object-model which does not exist anymore. Tough.

**Comment by eegli** [comment]
Ready to merge

**Pull request merged**